scalableminds/webknossos
# Make "Registering of All Segments for a BBox"- feature mag aware

### URL of deployed dev instance (used for testing):
- https://registerallsegmentsmagaware.webknossos.xyz/

### Steps to test:
- Open an annotation, ideally with multiple segmentation layers
- Open the bounding box tab and select "Register all segments for this bounding box"
- The feature should work as before, registering the segments into a new segment group
- Now use it on a very large BBox that exceeds the maximal volume in mag 1-1-1 (the limit is 512 * 512 * 512 Vx³)
- zoom out to coarser mags and try again to register all segments, until it is possible. This proves that the volume limit is taking the current mag into account
- Register the segments for a bounding box in multiple mags, from coarse to fine. In the corresponding segment groups, some segments from the finer mag should not be moved into the group that was created from the coarser mag. This would prove that the right mag is used to register the segments. (Context: If you register the segments for the same bbox in the same mag twice, two segment groups will be created. As segments can only be present in one group, each segment that is registered again will be moved into the latest "register all segments" segment group. So now, in our case, when a segment remains in the older group, this means that it was not registered again, and that means that the coarser mag was actually used)

- In an annotation with multiple segmentation layers, use this feature and make sure that the currently visible segmentation layer is used. One way to check this is to create a new volume annotation layer in an annotation with another segmentation layer, brush to create one segment, and then register the segments in both segmentation layers. For the newly created segmentation layer, only the brushed segment should be registered. In the other segmentation layer, this should be different.

### TODOs:
- [x] use current mag for volume limit
- [x] use segmentation in current mag to registering segments
- [x] use visible segmentation layer rather than first one. 
    - [x] test me!
    - [x] fix problem that segmentation layer name cannot be found sometimes -> I dont think this had anything todo with the new issue

### Issues:
- fixes #8044 

------
(Please delete unneeded items, merge only when none are left open)
